src/Microsoft.ML.Data/Transforms/HashTransform.cs

```diff
@@ -166,9 +166,9 @@ public void Save(ModelSaveContext ctx)
 
         private static string TestType(ColumnType type)
         {
-            if (type.ItemType.IsText || type.ItemType.IsKey)
+            if (type.ItemType.IsText || type.ItemType.IsKey || type.ItemType == NumberType.R4 || type.ItemType == NumberType.R8)
                 return null;
-            return "Expected Text or Key item type";
+            return "Expected Text, Key, Float or Double item type";
```

Comment:
>Float [](start = 40, length = 5)

This will not be familiar to people maybe say Single?

Reply:
I believe the recommendation is to use keywords in source code and official type names in messages/etc